resources/js/Components/BaseInput.vue

```diff
@@ -0,0 +1,44 @@
+<script setup lang="ts">
+
+type InputType = 'text' | 'email' | 'password' | 'number' | 'search';
+
+defineProps<{
+  modelValue: string
+  label?: string
+  type?: InputType
+  name: string
+  id: string
+  placeholder?: string
+  error?: string | null
+}>()
+
+defineEmits<{
+  'update:modelValue': [value: string]
+}>()
```

Comment:
Using `defineModel` would be a better idea. See: https://vuejs.org/guide/components/v-model.html#basic-usage